### DISCUSSION TITLE
Skip query option is  triggering a request

### DISCUSSION THREAD

**Discussion by robsontenorio** [Akryum/vue-apollo#388] [Q&A]
The same page/view handles `/users/<some-id>` (edit) and `/users/add` (add) routes. 

So, when some id is provided (ex `/users/99`) it will fetch the user data and works fine. But, when route is `/user/add` a request is sent to backend even with `skip` set. 

Am i missing something? Is something related to apollo/component life cycle?

```js
export default {
  apollo: {
    user: {
      query: findUserQuery,
      variables () {
        return { id: this.$route.params.id }
      }
    }
  },
  data () {
    return {
      user: {
           posts: []
           // more empty attributes in here
      }
    }
  },
  mounted () {
    if (this.$route.params.id === 'add') {
      // this will fetch data from backend
      this.$apollo.queries.user.skip = true
    }
  }
```

**Comment by GPrimola** [comment]
@robsontenorio I faced the same issue, but in fact you're missing something.
By the mounted lifecycle, apollo has already prepared the defied queries to be executed, so any changes in theirs variables would only take effect after que next tick.

I suggest you do the following:

```javascript
export default {
  apollo: {
    user: {
      query: findUserQuery,
      variables () {
        return { id: this.$route.params.id }
      },
      skip () {
        if (this.$route.params.id === 'add') {
          // this will fetch data from backend
          this.$apollo.queries.user.skip = true
        }
      }
    }
  },
  data () {
    return {
      user: {
           posts: []
           // more empty attributes in here
      }
    }
  }
```

That should work.
